lib/galaxy/tool_util/verify/asserts/__init__.py

```diff
@@ -44,10 +51,20 @@ def verify_assertion(data, assertion_description):
         raise AssertionError(errmsg)
 
     assert_function_args = getfullargspec(assert_function).args
+    assert_function_annotations = getfullargspec(assert_function).annotations
     args = {}
     for attribute, value in assertion_description["attributes"].items():
         if attribute in assert_function_args:
-            args[attribute] = value
+            # in python 3.10 issubclass(xyz, assert_function_annotations[attribute]) should work
+            types = assert_function_annotations[attribute]
+            if get_origin(assert_function_annotations[attribute]) is not None:
+                types = get_args(assert_function_annotations[attribute])
+            if issubclass(int, types):
+                args[attribute] = parse_bytesize(value)
```

Comment:
If someone specified `123M` for an integer value they'll get `123000000` as the value, or am I misunderstanding this ? Maybe you could narrow this down by creating a ByteCount type or similar ?

Reply:
Do you mean that we should not allow `123Mi` which would be `132070244352 = 123*1024^3`